### PR TITLE
Abuusama/dw 806/add input validation to

### PR DESCRIPTION
**Pull request by pythonpete32**
## Dework Task

https://app.dework.xyz/nation3/nation3-court-v1-p/view/board-l59hq64r?taskId=a9f092d3-bd8a-4cb6-9259-f22a61e88fec

## Related GitHub Issue

https://github.com/nation3/court/issues/61

## Screenshots (if appropriate):

CleanShot 2022-11-11 at 20 28 04

### CODE REVIEW

File: packages/ui-components/src/components/inputs/AddressInput.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		<div>
+			<input
+				type="text"
+				id="text-input"
+				className={clsx(
+					`bg-gray-50 text-gray-800 text-sm rounded-lg block w-full p-2.5`,
+					isValid
+						? `border border-gray-300 focus:border-${focusColor} focus:ring-${focusColor}`
+						: "border-solid ring-1 border-red-600 ring-red-600 focus:border-red-600 focus:ring-red-600",
+				)}
```
Comment (xPi2) [inline packages/ui-components/src/components/inputs/AddressInput.tsx:25]: This is cool! Could you do another PR to do this globally? There are a bunch of components that have a conditional className

File: packages/ui-components/src/utils/address.tsx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export const validateAddress = (address: string) => {
+	const regex = /^0x[a-fA-F0-9]{40}$/;
+	return regex.test(address);
+};
```
Comment (xPi2) [inline packages/ui-components/src/utils/address.tsx:4]: Use https://docs.ethers.io/v5/api/utils/address/#utils-isAddress instead, a valid address must also have a valid checksum.

File: packages/ui-components/src/components/inputs/AddressInput.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+				onBlur={(e) => checkAddress(e)}
```
Comment (xPi2) [inline packages/ui-components/src/components/inputs/AddressInput.tsx:26]: Remember that `props` could include an onBlur handler too. I would extend the handler with both the checkAddress() and onBlur() handler if provided.

File: packages/ui-components/src/components/inputs/AddressInput.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+export interface AddressInputProps extends InputHTMLAttributes<HTMLInputElement> {
+	focusColor?: string;
+}
+
+export const AddressInput = ({ focusColor = "bluesky", ...props }: AddressInputProps) => {
+	const [isValid, setIsValid] = React.useState(true);
+
+	const checkAddress = (value: ChangeEvent<HTMLInputElement>) => {
+		setIsValid(validateAddress(value.target.value));
+	};
```
Comment (xPi2) [inline packages/ui-components/src/components/inputs/AddressInput.tsx:14]: I suggest expecting the value directly, instead of the ChangeEvent

**Review by xPi2 — CHANGES_REQUESTED** [review]
Nice improvement!

File: packages/ui-components/src/components/inputs/AddressInput.tsx (hunk 1 above)
Comment (pythonpete32) [reply packages/ui-components/src/components/inputs/AddressInput.tsx:25]: will do

**Review by xPi2 — APPROVED** [review]
